test/integration/targets/lookup_sequence/tasks/main.yml

```
@@ -196,3 +196,91 @@
           - ansible_failed_result.msg == expected
       vars:
         expected: "bad formatting string: d"
+
+# Tests for lookup()/plugin() jinja invocation:
+# Many of these tests check edge case behaviors that are only possible when invoking query/lookup sequence through jinja.
+# While they aren't particularly intuitive, these tests ensure playbooks that could be relying on these behaviors don't
+# break in future
+- name: Test that sequence can be invoked using query/lookup with keyword args only
+  ping:
+    data: '{{ item }}'
+  loop: '{{ query("ansible.builtin.sequence", count=5, start=10) }}'
+  register: results1
+
+- name: Assert that sequence can be invoked using query/lookup with keyword args only
+  assert:
+    that:
+      - 'results1["results"][0]["ping"] == "10"'
+      - 'results1["results"][1]["ping"] == "11"'
+      - 'results1["results"][2]["ping"] == "12"'
+      - 'results1["results"][3]["ping"] == "13"'
+      - 'results1["results"][4]["ping"] == "14"'
```

There's no apparent reason to use this test structure. If it's actually doing something useful that should be explained in a comment. Otherwise, 
```suggestion
- name: Test lookup with keyword args only
  assert:
    that: '{{ query("ansible.builtin.sequence", count=5, start=10) == ["10", "11", "12", "13", "14"] }}'
```
Same for the other tests.

---

Thank you for demonstrating the better method! I'm still familiarizing myself with using Ansible's yaml and struggld to find a similar testing of query in the integration folder, so I essentially extrapolated that structure from some tests in the lookup_vars test module. I'll change all the test shortly. Appreciate the review.